src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -4786,6 +4823,7 @@ boolean executeReconnectCommand(TDSCommand newCommand) throws SQLServerException
              */
             boolean commandComplete = false;
             try {
+                newCommand.createCounter(null, activeConnectionProperties);
```

Comment:
What's this change for?

Reply:
It is added by Terry and as per his comment - _Deadloack and null counter TDSCommand fix_